### PR TITLE
[WIP] Implementation of the LCNN model

### PR DESCRIPTION
**Pull request by VIGNESHinZONE**
# Implementation of the LCNN Model

This the second PR on the implementation of the LCNN Model (in Pytorch and DGL). 
The featuriser for this model is implemented https://github.com/deepchem/deepchem/pull/2287 and I will add the doctest after this gets merged.

## Rsults
This model on average gets 0.43 RMSE error on the Platinum Adsorption Dataset(Consisting of 650 crystal adsorbate configuration). This model beats the accuracy proposed by the authors in the LCNN paper where they achieve 4.3 RMSE error given [here](https://pubs.acs.org/doi/10.1021/acs.jpcc.9b03370).

## Checklist

- [x] My code follows [the style guidelines of this project]
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/torch_models/lcnn.py [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+    energies.
+
+    The model follows the following steps
+
+    [1] It performs n lattice convolution operations.
+        For more details look at the LCNNBlock class
+    [2] Followed by Linear layer transforming into sitewise_n_feature
+    [3] Transformation to scalar value for each node.
+    [4] Average of properties per each element in a configuration
+
```
Comment (rbharath) [inline deepchem/models/torch_models/lcnn.py:268]: Can you add a usage example section to this docstring that shows how to use this model?

File: deepchem/models/torch_models/lcnn.py [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+    - model.fit(train, nb_epoch=10)
+
+    This model takes arbitrary configurations of Molecules on an adsorbate and predicts
+    their formation energy. These formation energies are found using DFT calculations and
+    LCNNModel is to automate that process. This model defines a crystal graph using the
+    distance between atoms. The crystal graph is an undirected regular graph (equal neighbours)
+    and different permutations of the neighbours are pre-computed using the LCNNFeaturizer.
+    On each node for each permutation, the neighbour nodes are concatenated which are further operated.
+    This model has only a node representation. Please confirm the detail algorithms from [1]_.
+
```
Comment (rbharath) [inline deepchem/models/torch_models/lcnn.py:370]: Can you add a usage example to this docstring?

File: deepchem/models/torch_models/lcnn.py [hunk 3]
```diff
@@ -0,0 +1,428 @@
+import torch
```
Comment (rbharath) [inline deepchem/models/torch_models/lcnn.py:1]: Can you add unit tests for `LCNN` and `LCNNModel`?

**Review by rbharath — COMMENTED** [review]
Thanks for the contribution! Did a first review pass. Can you add unit tests for `LCNN` and `LCNNModel`? I've also asked for a few fixes to formatting and improvements to the docstrings below

File: deepchem/models/torch_models/lcnn.py (hunk 2 above)
Comment (VIGNESHinZONE) [reply deepchem/models/torch_models/lcnn.py:370]: That can be added only after the LCNN featurisation gets merged otherwise, it might a give doctest error

**Comment by VIGNESHinZONE** [comment]
Thanks for review @rbharath, I can add the doctest & examples only after LCNNfeaturiser https://github.com/deepchem/deepchem/pull/2287 gets merged.

File: deepchem/models/torch_models/lcnn.py (hunk 2 above)
Comment (rbharath) [reply deepchem/models/torch_models/lcnn.py:370]: Ah good point. Ok will resolve this issue
